content/docs/operations/upgrade.md

```diff
@@ -65,16 +65,35 @@ During the upgrade, pods that run the new version of the control plane and proxi
 
 -->
 
-## Upgrade kgateway
+## Upgrade the Kubernetes Gateway API {#k8s-gw-api}
+
+Optionally, you can upgrade the Kubernetes Gateway API version. Make sure that the Kubernetes Gateway API version is compatible with the {{< reuse "docs/snippets/product-name.md" >}} version that you want to upgrade to.
```

Comment:
Why is upgrading k8s gw api optional? Shouldnt you be keeping it up with the latest version? or maybe i'm asking like, should we provide an opinionated upgrade path here so people know whether or not to upgrade k8s gw api? like "this documentation set uses the latest stable version of the k8s gw api, {{version}}. When you upgrade kgateway, check the version matrix (actually do we have a version matrix between kgateway and the k8s gateway api?). If a later version of the k8s gw api is available for the version of kgateway you are upgrading to, upgrading the k8s gw api is also recommended."